### PR TITLE
Enforces userId anonId requirement

### PR DESCRIPTION
**Pull request by benjaminhoffman**
addresses https://github.com/segmentio/analytics-ruby/issues/93

jira https://segment.atlassian.net/browse/EPD-376

why isn't there a test already in place for this condition?
@dominicbarnes @hankim813 @segmentio/platform-integrations

### CODE REVIEW

File: spec/segment/analytics/client_spec.rb [hunk 1]
```diff
@@ -99,6 +99,10 @@ class Analytics
           expect { client.identify({}) }.to raise_error(ArgumentError)
         end
 
+        it 'errors on an empty string' do
```
Comment (f2prateek) [inline spec/segment/analytics/client_spec.rb]: Also test for '' anonymousId?

**Comment by benjaminhoffman** [comment]
merging this even tho spec is failing.  it fails for an unrelated reason that should not be blocking this PR.  if we want, we can fix spec but that's out of scope.

**Comment by f2prateek** [comment]
I don't think jruby was the one failing. 2.0.0 is failing https://cloudup.com/ct7PFjQkATx

**Comment by benjaminhoffman** [comment]
@f2prateek correct.  spec is also failing.  there's a race condition or something funky going on, as well as an issue w jruby.  Both issues should be addressed but it's outside the scope of this PR.

**Pull request merged**